plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -509,7 +529,8 @@ public ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connecto
                 getColumns(icebergTable.schema(), typeManager),
                 icebergTable.location(),
                 getFileFormat(icebergTable),
-                icebergTable.properties());
+                icebergTable.properties(),
+                retryMode == RETRIES_ENABLED);
```

Comment:
`!= NO_RETRIES` for future compatibility.
or define a method with a switch statement forcing us to revisit the code if a new `io.trino.spi.connector.RetryMode` is added.